index.js

```diff
@@ -318,7 +318,8 @@ pip3 install -U virtualenv && ${runPy}
     this.log(ret.stderr.toString());
     this.log(ret.stdout.toString());
 
-    var isNotPipWarning = ret.stderr.indexOf('You are using pip version') < 0;
+    var isNotPipWarning = ret.stderr.indexOf('You are using pip version') < 0 &&
+                          ret.stderr.indexOf('DEPRECATION:') < 0;
     if (ret.error || (ret.stderr.length != 0 && isNotPipWarning)) {
       this.log('Unhandled error in pip, not deploying to AWS.');
       return BbPromise.reject(ret.error)
```

Comment:
One could consider to change the reject message for errors that is not a warning (undefined when from ret.stderr) .
return BbPromise.reject(ret.error || ret.stderr.toString())

Reply:
Ah, I briefly searched for how to differentiate errors from warnings in node.js but couldn't find the solution.
I guess it would be even better to not do the rejection for warnings at all?